### PR TITLE
create PR for DID Auth doc

### PR DESCRIPTION
**Pull request by sf316**
# Description

Added detailed doc on DID Auth under ./docs/dids folder and updated README.md. Please let me know if there is any more adjustment/improvement I could make. Besides the original issue, I also have two trivial questions:

1. What does generateJwt function do in src\bin\auth\agent\auth.agent.ts? I didn’t see anywhere else use access token generated from this one since we got generateDidJwt one.
2. (Not related) I think the following modules are missing for current front-end to function properly
- Bootstrap
- React-bootstrap
- React-toastify

## Related Issue

#677

### CODE REVIEW

File: examples/bri-3/docs/dids/did-authentication.md
Comment (skosito) [inline examples/bri-3/docs/dids/did-authentication.md:36]: since metamask is mentioned lets also add frontend we have to this flow explanation

File: examples/bri-3/docs/dids/did-authentication.md
Comment (skosito) [inline examples/bri-3/docs/dids/did-authentication.md:37]: there is no storing of did document as part of did auth flow

**Review by skosito — CHANGES_REQUESTED** [review]
awesome doc, thank you!
just some small comments before merge

File: examples/bri-3/docs/dids/did-authentication.md
Comment (Therecanbeonlyone1969) [inline examples/bri-3/docs/dids/did-authentication.md:5]:
```suggestion
[W3C Decentralized Identifiers](https://www.w3.org/TR/did-core/) (DIDs) are cryptographically variable, globally unique identifiers. By utilizing DIDs for authentication, they enable a decentralized and self-sovereign approach to digital identity, where users have control over their own identity data and can prove their identity without relying on centralized authorities.
```

File: examples/bri-3/docs/dids/did-authentication.md
Comment (Therecanbeonlyone1969) [reply examples/bri-3/docs/dids/did-authentication.md:36]: I agree

File: examples/bri-3/docs/dids/did-authentication.md
Comment (Therecanbeonlyone1969) [inline examples/bri-3/docs/dids/did-authentication.md:33]:
```suggestion
1. (Prerequisite) The BpiOperators first must configure their own .env files with the .env.sample and register BpiSubjects’ DID and public keys in the database
```

File: examples/bri-3/docs/dids/did-authentication.md
Comment (Therecanbeonlyone1969) [inline examples/bri-3/docs/dids/did-authentication.md:35]:
```suggestion
3. BpiOperator receives the response, resolves the DID to its DID Document and retrieves the key used from the `authentication` property, and then verifies the message signature with the `authentication` key. Then the BPI Operator generates an access (JWT) token for a successfully authorized account using BpiOperator’s private key.
```

File: examples/bri-3/docs/dids/did-authentication.md
Comment (Therecanbeonlyone1969) [inline examples/bri-3/docs/dids/did-authentication.md:37]:
```suggestion
5. BpiOperator receives the request, verifies the token, and, if successful routes the request to the target service. If the token verification fails (e.g. expiration), a failure message is generated, and the request is not routed to its target service.
```

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
See my suggestions. Great job otherwise.

File: examples/bri-3/docs/dids/did-authentication.md
Comment (biscuitdey) [inline examples/bri-3/docs/dids/did-authentication.md:17]: Please format this section so its easier to read. Suggestion: Italicise the example string and put the description on the next line.

File: examples/bri-3/docs/dids/did-authentication.md
Comment (biscuitdey) [inline examples/bri-3/docs/dids/did-authentication.md:15]: "Environment Variables" might be more appropriate heading as it will indicate readers to look for these variables in the .env file.

File: examples/bri-3/docs/dids/did-authentication.md
Comment (biscuitdey) [inline examples/bri-3/docs/dids/did-authentication.md:19]: Let's add how users can get these variable values (e.g. signup for Alchemy or get it from their metamask account, etc.).

**Review by biscuitdey — CHANGES_REQUESTED** [review]

**Comment by Kasshern** [comment]
Left some comments - mostly grammatical stuff. Will not block with request change. Already have two request changes blocking you. You can address these or not. Once others are satisfied and remove request changes I will approve.

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM now

**Review by biscuitdey — APPROVED** [review]
LGTM. Can be merged after other's comments are addressed.

**Comment by ognjenkurtic** [comment]
@sf316 please address the remaining comments and we can merge this one

**Comment by sf316** [comment]
> @sf316 please address the remaining comments and we can merge this one

@ognjenkurtic Done. Please review the changes, thank you!

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Comment by ognjenkurtic** [comment]
@skosito please remove request for change if everything is addressed

**Comment by ognjenkurtic** [comment]
> @skosito please remove request for change if everything is addressed

Forget about it, we just dismissed your review :D

**Review by Kasshern — APPROVED** [review]

**Pull request merged**
